### PR TITLE
Serialize Error Extensions in System.Text.Json with provided options

### PR DESCRIPTION
**Pull request by kyleadams-fm**
Extensions in errors should be serialized along with other values in the ExecutionResultJsonConverter

Addresses issue: https://github.com/graphql-dotnet/graphql-dotnet/issues/2552

### CODE REVIEW

**Comment by sungam3r** [comment]
@kyleadams-fm Thanks for fix. To confirm the error correction, we usually add a test. As I understand, it will not take much time. After that, I will be glad to take your change.

**Comment by kyleadams-fm** [comment]
Cheers @sungam3r, I've added a test for this case

**Review by Shane32 — APPROVED** [review]

**Comment by kyleadams-fm** [comment]
@sungam3r can I get approval on the workflows above? Looks like LGTM fails on all PRs when trying to build?

**Comment by Shane32** [comment]
> @sungam3r can I get approval on the workflows above? Looks like LGTM fails on all PRs when trying to build?

Yeah just ignore LGTM

**Pull request merged**
